glean-core/python/tests/metrics/test_timing_distribution.py

```diff
@@ -150,3 +150,57 @@ def override_time_ns():
 
         snapshot = metric.test_get_value()
         assert len(snapshot.values) < 318
+
+
+def test_measure(monkeypatch):
+    """
+    Test the TimingDistributionMetricType.measure context manager.
+    """
+    override_time = 0
+
+    def override_time_ns():
+        return override_time
+
+    monkeypatch.setattr(_util, "time_ns", override_time_ns)
+
+    metric = metrics.TimingDistributionMetricType(
+        disabled=False,
+        category="telemetry",
+        lifetime=Lifetime.APPLICATION,
+        name=f"timing_distribution",
+        send_in_pings=["baseline"],
+        time_unit=TimeUnit.NANOSECOND,
+    )
+
+    with metric.measure():
+        override_time = 1000
```

Comment:
First reading that I was slightly confused. Maybe add a comment taht we actively manipulate the timing function?